models/articles.model.js

```diff
@@ -88,3 +88,21 @@ exports.selectArticleComments = (article_id) => {
       return rows;
     });
 };
+
+exports.addComment = (newComment, article_id) => {
+  const { username, body } = newComment;
+
+  return db
+    .query(
+      `INSERT INTO comments(author, body, article_id) VALUES ($1, $2, $3) RETURNING *`,
+      [username, body, article_id]
+    )
+    .then(({ rows }) => {
+      return rows[0];
+    })
+    .catch((err) => {
+      if (err.code === "23503") {
+        return Promise.reject({ status: 404, msg: "Username not found" });
+      }
```

Comment:
Just have a catch in the controller - it's the end of the chain. 

Use next to send this error to an error handler in app.js and deal with it there.